pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -592,6 +596,10 @@ async def run_repomdrecord_download(name, location_href, downloader):
                     result = await downloader.run()
                     return name, location_href, result
 
+                file_extension = [
+                    record.location_href for record in repomd.records if record.type == "primary"
+                ][0].split(".")[-1]
```

Comment:
Just checking that the assumption is that the other.xml and the filelists.xml will have the same file_extension and ironically we are using primary.xml to identify that. Nothing wrong with that, and I think, it's a valid assumption, just checking my understanding.

Reply:
I agree, it's funny.  But yeah, they should all have the same type of compression.